### PR TITLE
Demonstrate behaviour of `SpecifierSet.__iter__`

### PR DESCRIPTION
**Pull request by hauntsaninja**
Linking #573

### CODE REVIEW

File: docs/specifiers.rst [hunk 1]
```diff
@@ -28,6 +28,10 @@ Usage
     >>> combined_spec &= "!=1.1"
     >>> combined_spec
     <SpecifierSet('!=1.1,>=1.0,~=1.0')>
+    >>> # We can iterate over the SpecifierSet to recover the
+    >>> # individual specifiers
+    >>> sorted(combined_spec, key=lambda x: x.operator)
```
Comment (pradyunsg) [inline docs/specifiers.rst:33]:
```suggestion
    >>> sorted(combined_spec, key=str)
```

**Review by pradyunsg — COMMENTED** [review]
Let's also include an example in the `__iter__` method's docstring.

File: docs/specifiers.rst (hunk 1 above)
Comment (hauntsaninja) [reply docs/specifiers.rst:33]: While key=str is a more minimal change to make doctests pass, it's also perhaps a little more inscrutable as a reader. Accessing `.operator` shows off some more of the API, so is maybe also more real-world-y.
(I don't feel strongly, so feel free to apply the change if you don't agree)

**Comment by hauntsaninja** [comment]
Oh, I was confused about what you meant by docstring, but now I see all your changes on main :-)

File: docs/specifiers.rst (hunk 1 above)
Comment (uranusjr) [reply docs/specifiers.rst:35]: Sorting by operator does not really make sense though. Personally I prefer `str` here.

Comment (uranusjr) [reply docs/specifiers.rst:35]: If you really really want to show off the API, sorting by version is probably more meaningful (but the syntax could be a bit too verbose for a short example).

Comment (hauntsaninja) [reply docs/specifiers.rst]: Great, made the change!

**Comment by pradyunsg** [comment]
Thanks @hauntsaninja! ^>^

**Pull request merged**